src/main/java/net/snowflake/ingest/streaming/internal/IcebergParquetValueParser.java

```diff
@@ -430,28 +449,71 @@ private static ParquetBufferValue getStructValue(
       SubColumnFinder subColumnFinder,
       ZoneId defaultTimezone,
       final long insertRowsCurrIndex,
-      boolean isDescendantsOfRepeatingGroup) {
+      boolean isDescendantsOfRepeatingGroup,
+      InsertValidationResponse.InsertError error) {
     Map<String, ?> structVal =
         DataValidationUtil.validateAndParseIcebergStruct(
             subColumnFinder.getDotPath(type.getId().toString()), value, insertRowsCurrIndex);
     Set<String> extraFields = new HashSet<>(structVal.keySet());
+    List<String> missingFields = new ArrayList<>();
     List<Object> listVal = new ArrayList<>(type.getFieldCount());
     float estimatedParquetSize = 0f;
     for (int i = 0; i < type.getFieldCount(); i++) {
-      ParquetBufferValue parsedValue =
-          parseColumnValueToParquet(
-              structVal.getOrDefault(type.getFieldName(i), null),
-              type.getType(i),
-              statsMap,
-              subColumnFinder,
-              defaultTimezone,
-              insertRowsCurrIndex,
-              isDescendantsOfRepeatingGroup);
       extraFields.remove(type.getFieldName(i));
-      listVal.add(parsedValue.getValue());
-      estimatedParquetSize += parsedValue.getSize();
+      if (structVal.containsKey(type.getFieldName(i))) {
+        ParquetBufferValue parsedValue =
+            parseColumnValueToParquet(
+                structVal.get(type.getFieldName(i)),
+                type.getType(i),
+                statsMap,
+                subColumnFinder,
+                defaultTimezone,
+                insertRowsCurrIndex,
+                isDescendantsOfRepeatingGroup,
+                error);
+        listVal.add(parsedValue.getValue());
+        estimatedParquetSize += parsedValue.getSize();
+      } else {
+        if (type.getType(i).isRepetition(Repetition.REQUIRED)) {
+          missingFields.add(type.getFieldName(i));
+        } else {
+          listVal.add(null);
+        }
+      }
+    }
+
+    if (!missingFields.isEmpty()) {
+      missingFields.forEach(
```

Comment:
please use regular for loop to avoid the unnecessary indentation / lambda!

Reply:
Changed to native for loop.